website/content/docs/k8s/connect/ingress-controllers.mdx

```diff
@@ -0,0 +1,74 @@
+---
+layout: docs
+page_title: Ingress Controller Integrations
+description: Configuring Ingress Controllers With Consul On Kubernetes
+---
+
+# Configuring Ingress Controllers with Consul on Kubernetes
+
+-> This topic requires familiarity with [Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)
+
+-> The following information requires Consul 1.10+, Consul-k8s 0.26+, Consul-helm 0.32+ configured with
+[Transparent Proxy](/docs/connect/transparent-proxy) mode enabled.
+
+This page describes a general approach for integrating Ingress Controllers with Consul on Kubernetes to secure traffic from the Controller
+to the backend services. This allows Consul to transparently manage secure traffic from the ingress point through the entire traffic flow of the service.
+
+A few steps are generally required to enable an Ingress controller to join the mesh and pass traffic through to a service:
+
+* Enable connect-injection via an annotation on the Ingress Controller's deployment: `consul.hashicorp.com/connect-inject` is `true`.
+
+* Using the following annotations on the Ingress controller's deployment, setup exclusion rules for it's ports.
+  * `consul.hashicorp.com/transparent-proxy-exclude-inbound-ports` - Provides the ability to exclude a list of ports for
+inbound traffic that the service exposes from redirection. Typical configurations would require all inbound service ports
+for the controller to be included in this list.
+  * `consul.hashicorp.com/transparent-proxy-exclude-outbound-ports` - Provides the ability to exclude a list of ports for
+outbound traffic that the service exposes from redirection.
+  * `consul.hashicorp.com/transparent-proxy-exclude-outbound-cidrs` - Provides the ability to exclude a list of CIDRs that
+the service communicates with for outbound requests from redirection. It is somewhat common that an Ingress controller
+will expect to make API calls to the Kubernetes service for service/endpoint management. As such including the ClusterIP of the
+Kubernetes service is common.
+
+```yaml
+# An Example configuration showing some typical podAnnotations settings
+deployment:
+  podAnnotations:
```

Comment:
This syntax seems specific to a particular Helm chart. Should this example be changed to show how to add annotations to a generic Deployment resource?

Reply:
the idea was that we were hoping that it could be inferred via most of the helm charts having a `podAnnotations` section or `deployment` section, but yeah this is actually a traefik example, I'm struggling to find a more generic way to visualize it though. Any ideas?

Reply:
@Blake This is the particular pattern that I've seen most Ingress Controllers leverage annotations for their underlying pods and we know is tested and working.

Reply:
It think the best thing we can do is add a comment inline that this example is specific to Traefik. At the very least, someone will examine this a few times before copying and pasting it.

Reply:
@david-yu My concern is that this configuration example is not a generic example on how to apply annotations to a Deployment. Its is a specific example on how to configure annotations for Traefik.

If I look at the Helm charts for a few of the top 5 ingress controller's (according to the [CNCF's 2020 Survey](https://www.cncf.io/wp-content/uploads/2020/11/CNCF_Survey_Report_2020.pdf)), they each provide different methods for configuring annotations on the pods of a deployment.

**NGINX**

```yaml
controller:
  pod:
    annotations: {}
```

**HAProxy** (Kong also uses this)

```yaml
podAnnotations: {}
```
**Traefik**

```yaml
deployment:
  podAnnotations: {}
```

I think it would be helpful to either document how to annotate a generic Deployment object, or add a sentence stating "The following example shows how to configure Pod annotations in the Traefik Ingress controller Helm chart. Refer to the documentation for your specific ingress on how to configure pod annotations for the ingress controller."